haas/rest.py

```diff
@@ -186,18 +195,35 @@ def _do_validation(schema, kwargs):
     `ValidationError`.
     """
 
-    if flask.request.data == '':
-        # No request body
-        final_kwargs = {}
+    final_kwargs = {}
+
+    if flask.request.method == "GET":
+        # GET requests can use URL and query parameter arguments
+        if flask.request.data != '':
+            raise ValidationError("GET request made with a non-empty request"
+                                  " body")
+
+        for key,value in flask.request.args.iteritems():
+            if final_kwargs.has_key(key):
+                raise ValidationError("Parameter specified more than once")
+            elif value is None or value == '':
+                # TODO: if we want to take flags (ie - an option that has no
+                # value), change this check and add logic in the next else to
+                # take care of that. We'll also need to pick a standard value
+                # to represent a flag (like 'True', 'None' or '').
+                raise ValidationError("Empty parameter specified")
```

Comment:
With JSON, it's not really an issue since the caller explicitly sets a value and our verification system checks its type (or potentially could even check values). So if the API call expects `null` as the "flag", that's ok.

As the comment mentions, we'd need to adopt a standard for GETs, and I figured that we can do that if the need arises.

Do you feel strongly that we should pick something now? If so, which value do you propose?